design/edit-existing-release-config.md

```diff
@@ -0,0 +1,98 @@
+# Allow editing and deploying existing application version.
+
+Once an application is installed, Admin Console will automatically check for updates, and these updates will be automatically added to version history but not deployed.
+Admin Console only allows updating configuration of the latest release on this list, which is not the one that is deployed.
+In order to update the currently running version, vendors have to move license to a new channel and promote that release to the new channel.
+
+## Goals
+
+- Allow deploying any application version with updated configuration.
+- Preserve configuration change history.
+- Preserve ability to calculate YAML diff due to config changes.
+- Keep the number of application versions managebale.
+- Keep the complexity of config update manageable.
+Right now only one version archive needs to be read and one archive needs to be written.
+
+## Non Goals
+
+- UI design for viewing config change history.
+- UI design for viewing YAML diff for config changes.
+- Rolling back deployed changes.
+This can be achieved with:
+  - `kots download`/`kots upload` commands
+  - by updating config again using previous value(s).
+
+## Background
+
+In some cases application upgrades must go through a process that can take weeks.
+However during this time an application configuration may need to be updated.
+
+## High-Level Design
+
+When application configuration is updated, the `upstream` spec does not change.
+Only the contents of the `upstream/userdata/config.yaml` are changed.
+The contents of the `base` spec are then generated from `upstream` and `config.yaml`.
+The proposal is to allow modifying application version archives and versioning `config.yaml` within the archive.
+
+## Detailed Design
+
+1. `ConfigValues` will be versioned within the `upstream/userdata/` directory.
+   - Config version sequence will be zero-based.
+   - Annotation `kots.io/config-sequence` will be added to ConfigValues spec.
+   - Current configuration will be stored in `upstream/userdata/config.yaml` (same file as now)
+   - **All** configurations will be stored in files with version numbers included in the files: `config-N.yaml`
```

Comment:
This feels _slightly_ brittle. Sequencing things is hard and if there's a gap, things will break? An alternative might be to make config.yaml a multi-doc yaml, with each version ORDERED with the newest at the bottom?

Reply:
I'd like to keep keep config.yaml as is because it's an easy place to copy config from to use with kots CLI.
How about adding config-history.yaml, which will have a new kind: an array of configs.